### PR TITLE
precompile_headers all platforms

### PR DESCRIPTION
**Pull request by crazyhappygame**
THIS PROJECT IS IN DEVELOPMENT MODE. Any pull requests should merge to branch `dev`, otherwise will be rejected immediately.

## Describe your changes
precompile_headers all platforms
## Checklist before requesting a review
-  [x] I have performed a self-review of my code.
-  [ ] If it is a core feature, I have added thorough tests.
-  [x] I have checked readme and add important infos to this PR (if it needed).
-  [x] An improved cpp-test/lua-test is not needed (explain why not, thanks).

### CODE REVIEW

**Comment by halx99** [comment]
Please check the ios/tvos ci check failed.

File: core/CMakeLists.txt [hunk 1]
```diff
@@ -157,16 +157,10 @@ if(XCODE OR VS)
     ax_mark_code_files("${_AX_CORE_LIB}")
 endif()
 
+message("CMake ${_AX_CORE_LIB} target_precompile_headers")
+target_precompile_headers(${_AX_CORE_LIB} PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:cocos2d.h>")
+
 if(WINDOWS)
-    # precompiled header. Compilation time speedup ~4x, but don't know how Ninja to support PCH yet and needs help from someone.
-    if(NOT CMAKE_GENERATOR STREQUAL "Ninja")
```
Comment (halx99) [inline core/CMakeLists.txt:162]: Ninja generator was borken?

Comment (crazyhappygame) [reply core/CMakeLists.txt:162]:
```
if(NOT CMAKE_GENERATOR STREQUAL "Ninja")
```
code was added by me in cocos2d-x  3.
I tried add support for Windows and Ninja. Currently I believe that for windows we should support only msvc compiler. 
Not many people use Ninja on windows for axmol/cocos2d-x - many things breaks...

Comment (crazyhappygame) [reply core/CMakeLists.txt:162]: Can we merge it?
It speedup other platforms builds by ~ 30%?

**Comment by halx99** [comment]
since vs2019, user can use vs open folder to open engine root directory, and vs will auto generate cmake ninja build files, then choud start run and debug tests.

File: core/CMakeLists.txt (hunk 1 above)
Comment (halx99) [reply core/CMakeLists.txt:162]: lgtm except Ninja generator was broken

**Comment by halx99** [comment]
This pr cause some issues like: #914, if no idea to sove it, I will revert first.

**Comment by crazyhappygame** [comment]
GitHub actions are green.
I think what is needed is to clean/remove and create new cmake configuration

**Comment by halx99** [comment]
Idelete whole build folder and regenerate, still fail on my local machine (cmake 3.23.2, 3.24.2 + vs2022 17.3.5)

**Comment by halx99** [comment]
@crazyhappygame still has config error in Android Studio:
```txt
HelloCpp.HelloCpp.main | debug | arm64-v8a Compiler exited with error code 1: D:\dev\adt\sdk\ndk\23.2.8568313\toolchains\llvm\prebuilt\windows-x86_64\bin\clang++.exe -xc++ --target=aarch64-none-linux-android21 --gcc-toolchain=D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64 --sysroot=D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/sysroot -DANDROID -DASTCENC_AVX=0 -DASTCENC_F16C=0 -DASTCENC_NEON=1 -DASTCENC_POPCNT=0 -DASTCENC_SSE=0 -DAX_USE_ALSOFT=1 -DBT_USE_SSE_IN_API=1 -DCP_USE_CGTYPES=0 -DCP_USE_DOUBLES=0 -DFMT_HEADER_ONLY=1 -DLLHTTP_STRICT_MODE=0 -DOPENSSL_SUPPRESS_DEPRECATED=1 -DP2T_STATIC_EXPORTS -DUSE_FILE32API -D_AX_DEBUG=1 -ID:/dev/axmol1/axmol -ID:/dev/axmol1/axmol/thirdparty -ID:/dev/axmol1/axmol/extensions -ID:/dev/axmol1/axmol/core -ID:/dev/axmol1/axmol/core/platform -ID:/dev/axmol1/axmol/thirdparty/robin-map/include -ID:/dev/axmol1/axmol/thirdparty/jni.hpp/include -ID:/dev/axmol1/axmol/thirdparty/freetype/include -ID:/dev/axmol1/axmol/thirdparty/fmt/include -ID:/dev/axmol1/axmol/thirdparty/png -ID:/dev/adt/sdk/ndk/23.2.8568313/sources/android/cpufeatures -ID:/dev/axmol1/axmol/thirdparty/box2d/include -ID:/dev/axmol1/axmol/thirdparty/chipmunk/include -IE:/dev/projects/HelloCpp/proj.android/app/.cxx/Debug/54m2x4as/arm64-v8a/engine/thirdparty/freetype/include -ID:/dev/axmol1/axmol/thirdparty/recast/.. -ID:/dev/axmol1/axmol/thirdparty/bullet/. -ID:/dev/axmol1/axmol/thirdparty/webp/src/webp -ID:/dev/axmol1/axmol/thirdparty/pugixml -ID:/dev/axmol1/axmol/thirdparty/xxhash/. -ID:/dev/axmol1/axmol/thirdparty/lz4/. -ID:/dev/axmol1/axmol/thirdparty/clipper2/. -ID:/dev/axmol1/axmol/thirdparty/ConvertUTF/. -ID:/dev/axmol1/axmol/thirdparty/poly2tri/.. -ID:/dev/axmol1/axmol/thirdparty/astc/.. -ID:/dev/axmol1/axmol/thirdparty/openal/include -ID:/dev/axmol1/axmol/thirdparty/openal/include/AL -ID:/dev/axmol1/axmol/thirdparty/ogg/include -ID:/dev/axmol1/axmol/thirdparty/unzip/. -ID:/dev/axmol1/axmol/thirdparty/llhttp/include -isystem D:/dev/axmol1/axmol/thirdparty/zlib/include -isystem D:/dev/axmol1/axmol/thirdparty/jpeg-turbo/include -isystem D:/dev/axmol1/axmol/thirdparty/openssl/include -isystem D:/dev/axmol1/axmol/thirdparty/curl/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -frtti -fexceptions -fsigned-char -fno-limit-debug-info -fPIC -std=c++20 -Winvalid-pch -Xclang -include-pch -Xclang E:/dev/projects/HelloCpp/proj.android/app/.cxx/Debug/54m2x4as/arm64-v8a/engine/axmol/core/CMakeFiles/axmol.dir/cmake_pch.hxx.pch -Xclang -include -Xclang E:/dev/projects/HelloCpp/proj.android/app/.cxx/Debug/54m2x4as/arm64-v8a/engine/axmol/core/CMakeFiles/axmol.dir/cmake_pch.hxx -fpch-preprocess -v -dD -E Android (8481493, based on r416183c2) clang version 12.0.9 (https://android.googlesource.com/toolchain/llvm-project c935d99d7cf2016289302412d708641d52d2f7ee) Target: aarch64-none-linux-android21 Thread model: posix InstalledDir: D:\dev\adt\sdk\ndk\23.2.8568313\toolchains\llvm\prebuilt\windows-x86_64\bin Found candidate GCC installation: D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/lib/gcc/aarch64-linux-android\4.9.x Selected GCC installation: D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/lib/gcc/aarch64-linux-android/4.9.x Candidate multilib: .;@m64 Selected multilib: .;@m64 (in-process) "D:\\dev\\adt\\sdk\\ndk\\23.2.8568313\\toolchains\\llvm\\prebuilt\\windows-x86_64\\bin\\clang++.exe" -cc1 -triple aarch64-none-linux-android21 -E -disable-free -disable-llvm-verifier -discard-value-names -main-file-name compiler-file4786404273739463101 -mrelocation-model pic -pic-level 2 -fhalf-no-semantic-interposition -mframe-pointer=non-leaf -fno-rounding-math -mconstructor-aliases -munwind-tables -target-cpu generic -target-feature +neon -target-abi aapcs -mllvm -aarch64-fix-cortex-a53-835769=1 -fallow-half-arguments-and-returns -fno-split-dwarf-inlining -debug-info-kind=standalone -dwarf-version=4 -debugger-tuning=gdb -v -ffunction-sections -fdata-sections -resource-dir "D:\\dev\\adt\\sdk\\ndk\\23.2.8568313\\toolchains\\llvm\\prebuilt\\windows-x86_64\\lib64\\clang\\12.0.9" -isystem D:/dev/axmol1/axmol/thirdparty/zlib/include -isystem D:/dev/axmol1/axmol/thirdparty/jpeg-turbo/include -isystem D:/dev/axmol1/axmol/thirdparty/openssl/include -isystem D:/dev/axmol1/axmol/thirdparty/curl/include -D ANDROID -D ASTCENC_AVX=0 -D ASTCENC_F16C=0 -D ASTCENC_NEON=1 -D ASTCENC_POPCNT=0 -D ASTCENC_SSE=0 -D AX_USE_ALSOFT=1 -D BT_USE_SSE_IN_API=1 -D CP_USE_CGTYPES=0 -D CP_USE_DOUBLES=0 -D FMT_HEADER_ONLY=1 -D LLHTTP_STRICT_MODE=0 -D OPENSSL_SUPPRESS_DEPRECATED=1 -D P2T_STATIC_EXPORTS -D USE_FILE32API -D _AX_DEBUG=1 -I D:/dev/axmol1/axmol -I D:/dev/axmol1/axmol/thirdparty -I D:/dev/axmol1/axmol/extensions -I D:/dev/axmol1/axmol/core -I D:/dev/axmol1/axmol/core/platform -I D:/dev/axmol1/axmol/thirdparty/robin-map/include -I D:/dev/axmol1/axmol/thirdparty/jni.hpp/include -I D:/dev/axmol1/axmol/thirdparty/freetype/include -I D:/dev/axmol1/axmol/thirdparty/fmt/include -I D:/dev/axmol1/axmol/thirdparty/png -I D:/dev/adt/sdk/ndk/23.2.8568313/sources/android/cpufeatures -I D:/dev/axmol1/axmol/thirdparty/box2d/include -I D:/dev/axmol1/axmol/thirdparty/chipmunk/include -I E:/dev/projects/HelloCpp/proj.android/app/.cxx/Debug/54m2x4as/arm64-v8a/engine/thirdparty/freetype/include -I D:/dev/axmol1/axmol/thirdparty/recast/.. -I D:/dev/axmol1/axmol/thirdparty/bullet/. -I D:/dev/axmol1/axmol/thirdparty/webp/src/webp -I D:/dev/axmol1/axmol/thirdparty/pugixml -I D:/dev/axmol1/axmol/thirdparty/xxhash/. -I D:/dev/axmol1/axmol/thirdparty/lz4/. -I D:/dev/axmol1/axmol/thirdparty/clipper2/. -I D:/dev/axmol1/axmol/thirdparty/ConvertUTF/. -I D:/dev/axmol1/axmol/thirdparty/poly2tri/.. -I D:/dev/axmol1/axmol/thirdparty/astc/.. -I D:/dev/axmol1/axmol/thirdparty/openal/include -I D:/dev/axmol1/axmol/thirdparty/openal/include/AL -I D:/dev/axmol1/axmol/thirdparty/ogg/include -I D:/dev/axmol1/axmol/thirdparty/unzip/. -I D:/dev/axmol1/axmol/thirdparty/llhttp/include -D ANDROID -D _FORTIFY_SOURCE=2 -D ___CIDR_DEFINITIONS_END -isysroot D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/sysroot -internal-isystem D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1 -internal-isystem D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/local/include -internal-isystem "D:\\dev\\adt\\sdk\\ndk\\23.2.8568313\\toolchains\\llvm\\prebuilt\\windows-x86_64\\lib64\\clang\\12.0.9\\include" -internal-externc-isystem D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/aarch64-linux-android -internal-externc-isystem D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/sysroot/include -internal-externc-isystem D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include -Wformat -Winvalid-pch -std=c++20 -fdeprecated-macro -fdebug-compilation-dir "E:\\dev\\projects\\HelloCpp\\proj.android\\app\\.cxx\\Debug\\54m2x4as\\arm64-v8a" -ferror-limit 19 -stack-protector 2 -fgnuc-version=4.2.1 -fno-implicit-modules -fcxx-exceptions -fexceptions -dD -include-pch E:/dev/projects/HelloCpp/proj.android/app/.cxx/Debug/54m2x4as/arm64-v8a/engine/axmol/core/CMakeFiles/axmol.dir/cmake_pch.hxx.pch -include E:/dev/projects/HelloCpp/proj.android/app/.cxx/Debug/54m2x4as/arm64-v8a/engine/axmol/core/CMakeFiles/axmol.dir/cmake_pch.hxx -o - -x c++ "C:\\Users\\halx99\\AppData\\Local\\Temp\\compiler-file4786404273739463101" clang -cc1 version 12.0.9 based upon LLVM 12.0.9git default target x86_64-w64-windows-gnu error: unable to read PCH file E:/dev/projects/HelloCpp/proj.android/app/.cxx/Debug/54m2x4as/arm64-v8a/engine/axmol/core/CMakeFiles/axmol.dir/cmake_pch.hxx.pch: 'No such file or directory' ignoring nonexistent directory "D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/sysroot/include" ignoring duplicate directory "D:/dev/axmol1/axmol/thirdparty" ignoring duplicate directory "D:/dev/axmol1/axmol/thirdparty" ignoring duplicate directory "D:/dev/axmol1/axmol/thirdparty" #include "..." search starts here: #include ...> search starts here: D:/dev/axmol1/axmol D:/dev/axmol1/axmol/thirdparty D:/dev/axmol1/axmol/extensions D:/dev/axmol1/axmol/core D:/dev/axmol1/axmol/core/platform D:/dev/axmol1/axmol/thirdparty/robin-map/include D:/dev/axmol1/axmol/thirdparty/jni.hpp/include D:/dev/axmol1/axmol/thirdparty/freetype/include D:/dev/axmol1/axmol/thirdparty/fmt/include D:/dev/axmol1/axmol/thirdparty/png D:/dev/adt/sdk/ndk/23.2.8568313/sources/android/cpufeatures D:/dev/axmol1/axmol/thirdparty/box2d/include D:/dev/axmol1/axmol/thirdparty/chipmunk/include E:/dev/projects/HelloCpp/proj.android/app/.cxx/Debug/54m2x4as/arm64-v8a/engine/thirdparty/freetype/include D:/dev/axmol1/axmol/thirdparty/bullet/. D:/dev/axmol1/axmol/thirdparty/webp/src/webp D:/dev/axmol1/axmol/thirdparty/pugixml D:/dev/axmol1/axmol/thirdparty/xxhash/. D:/dev/axmol1/axmol/thirdparty/lz4/. D:/dev/axmol1/axmol/thirdparty/clipper2/. D:/dev/axmol1/axmol/thirdparty/ConvertUTF/. D:/dev/axmol1/axmol/thirdparty/openal/include D:/dev/axmol1/axmol/thirdparty/openal/include/AL D:/dev/axmol1/axmol/thirdparty/ogg/include D:/dev/axmol1/axmol/thirdparty/unzip/. D:/dev/axmol1/axmol/thirdparty/llhttp/include D:/dev/axmol1/axmol/thirdparty/zlib/include D:/dev/axmol1/axmol/thirdparty/jpeg-turbo/include D:/dev/axmol1/axmol/thirdparty/openssl/include D:/dev/axmol1/axmol/thirdparty/curl/include D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/c++/v1 D:/dev/adt/sdk/ndk/23.2.8568313/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/local/include
```

**Pull request merged**